### PR TITLE
AMLS-4832 | Adding business name and services to renewal

### PR DESCRIPTION
**Pull request by owen-opencast**
Added business name and services list to renewal's About your Business page. The link under the services list is hidden for renewal.

## How Has This Been Tested?

Unit testing, passes acceptance tests.

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [ ] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: app/views/include/services_sidebar.scala.html [hunk 1]
```diff
@@ -39,10 +40,13 @@ <h2 class="heading-small">Your services</h2>
                     @serviceNames.map(name => Html("<li>" + name + "</li>"))
                 </ul>
             }
+
+            @if(displayLink) {
             <p class="sidebar-top-spacing">
```
Comment (drewmiley) [inline app/views/include/services_sidebar.scala.html]: Indent this?

File: app/views/registrationamendment/registration_amendment.scala.html [hunk 2]
```diff
@@ -48,7 +48,7 @@
     </div>
 
     @* Sidebar *@
-    @registration_progress_sidebar(canEditPreApplication, businessName, serviceNames)
+    @services_sidebar(canEditPreApplication, true, businessName, serviceNames)
```
Comment (drewmiley) [inline app/views/registrationamendment/registration_amendment.scala.html]: Name boolean param

File: app/views/registrationprogress/registration_progress.scala.html [hunk 3]
```diff
@@ -46,7 +46,7 @@
     </div>
 
     @* registration progress *@
-    @registration_progress_sidebar(canEditPreApplication, businessName, serviceNames)
+    @services_sidebar(canEditPreApplication, true, businessName, serviceNames)
```
Comment (drewmiley) [inline app/views/registrationprogress/registration_progress.scala.html]: Name boolean param

File: app/views/renewal/renewal_progress.scala.html [hunk 4]
```diff
@@ -76,6 +78,10 @@
         </div>
     </div>
 
+    <div class="grid-layout grid-layout--stacked">
+        @services_sidebar(false, false, businessName, serviceNames)
```
Comment (drewmiley) [inline app/views/renewal/renewal_progress.scala.html]: Name boolean params

File: test/views/registrationprogress/registration_progressSpec.scala [hunk 5]
```diff
@@ -47,5 +50,19 @@ class registration_progressSpec extends AmlsSpec with MockitoSugar with AddressG
 
       doc.select("h2.heading-small").first().ownText() must be("Your business")
     }
+
+    "show the business name and services" in new ViewFixture {
+      def view = views.html.registrationprogress.registration_progress(sections, true, businessName, serviceNames, true)
+      val element = doc.getElementsByClass("business-info").first()
+      element.text() must include { serviceNames.head }
```
Comment (drewmiley) [inline test/views/registrationprogress/registration_progressSpec.scala]: You could loop through the service names and assert within the forEach

File: test/views/renewal/renewal_progressSpec.scala [hunk 6]
```diff
@@ -49,8 +50,31 @@ class renewal_progressSpec extends AmlsSpec with MustMatchers{
...
+
+      override def view = views.html.renewal.renewal_progress(Seq.empty, businessName, serviceNames, false, true, readyForRenewal)
+
+      html must include (Messages("renewal.progress.intro", DateHelper.formatDate(renewalDate)).convertLineBreaks)
+    }
+
+    "show the business name and services" in new ViewFixture {
+      override def view = views.html.renewal.renewal_progress(Seq.empty, businessName, serviceNames, false, true, readyForRenewal)
+      val element = doc.getElementsByClass("business-info").first()
+      element.text() must include { serviceNames.head }
```
Comment (drewmiley) [inline test/views/renewal/renewal_progressSpec.scala]: You could loop through the service names and assert within the forEach

File: app/controllers/renewal/RenewalProgressController.scala [hunk 7]
```diff
@@ -59,12 +59,13 @@ class RenewalProgressController @Inject()
               cache <- OptionT(dataCacheConnector.fetchAll)
               businessMatching <- OptionT.fromOption[Future](cache.getEntry[BusinessMatching](BusinessMatching.key))
             } yield {
-
+              val businessName = businessMatching.reviewDetails.map(r => r.businessName).getOrElse("")
+              val activities = businessMatching.activities.fold(Seq.empty[String])(_.businessActivities.map(_.getMessage()).toSeq)
```
Comment (drewmiley) [inline app/controllers/renewal/RenewalProgressController.scala]: Do we need the toSeq call? I suspect we do, but just to clarify.

Comment (owen-opencast) [reply app/controllers/renewal/RenewalProgressController.scala]: It does currently, the template for services_sidebar was pre-existing and takes a Seq.

Comment (drewmiley) [reply app/controllers/renewal/RenewalProgressController.scala]: A most acceptable answer

**Pull request merged**